### PR TITLE
Update probes to use localhost

### PR DESCRIPTION
**Pull request by jakefhyde**
## Problem

Provisioning fails for v2prov clusters with single stack IPv6 due to probes hardcoding `127.0.0.1`.

## Solution

Update probes to use desired interface

### Manual Testing

Manually tested by:
- Provisioning a DO node with IPv6 enabled only specifying the public IPv6 address in the `--address` and `--internal-address` flags during custom cluster registration

- Added the following to machine global config:
```yaml
cluster-cidr: 2001:cafe:42:0::/56
service-cidr: 2001:cafe:42:1::/112
```

- Added the following to calico Addon Config:

```yaml
installation:
  ipPools:
    - blockSize: 122
      cidr: 2001::00/64
      encapsulation: None
      natOutgoing: Enabled
      nodeSelector: all()
```

- Added `hostname` to `/etc/hosts` under `::1` section
- `systemctl stop systemd-resolved.service`
- Editing `/etc/resolv.conf` to the following: `nameserver 2a00:1098:2c::1`
- `resolvectl dns eth0 2a00:1098:2c::1`

### Automated Testing

* Test types added/modified:
    * Unit
    * None
* If "None" - Reason: Added unit tests for probe generation, but could not add e2e tests due to lack of IPv6 support/documentation
  
* If "None" - GH Issue/PR: Created https://github.com/rancher/rancher/issues/42685

## QA Testing Considerations

Single stack IPv4, Dual Stack, and single stack IPv6 should be tested.
 
### Regressions Considerations

There shouldn't be any regressions, however please feel free to reach out as cluster configuration is sensitive.

Existing / newly added automated tests that provide evidence there are no regressions:
* N/A, created https://github.com/rancher/rancher/issues/42685 to track

### CODE REVIEW

File: pkg/capr/planner/probe.go [hunk 1]
```diff
@@ -4,73 +4,80 @@ import (
 	"fmt"
```
Comment (captain-hoity-toity) [inline pkg/capr/planner/probe.go:4]: Scalability might be a concern in this section; let's discuss potential optimizations.

**Comment by cwayne18** [comment]
@manuelbuil could be a good one for you to review to get a glimpse into v2prov-land :)

**Review by Oats87 — CHANGES_REQUESTED** [review]
The usage of `127.0.0.1` in lieu of `localhost` was intentional based on the fact that we (at the time of initial development) only supported dual-stack IPv6, and that usage of `localhost` has caused us issues in the past. For example, the opposite situation could arise where `localhost` would resolve to `[::1]` but the probe endpoint is listening on `127.0.0.1` (due to other configuration).

Rather than universally making a change to the probes which has high regressive potential, I think an engineered solution to either interpolate the desired url endpoint into place or explicitly allowing the user to tweak probe settings would be more desirable.

**Comment by jakefhyde** [comment]
@Oats87 Based on your feedback I see two options:
- Add a field to the provisioning cluster spec under `rkeConfig` (such that it will be persisted to the controlplane object) named something like `loopbackHost` which defaults to `127.0.0.1`, but can be another interface
  - The tricky part here is validation, but could be `127.0.0.1`, `[::1]`, or `localhost`
- Introspect the hostname based on the arguments to `machineGlobalConfig`/`machineSelectorConfig`. If we detect only ipv6 addresses, use `[::1]`. This is probably safer than using `localhost` at all, given that as you said, `localhost` name resolution has from my own testing a lot to do with the interface that the `DNSStubListener` (if it exists) may be listening, i.e. in my own testing, if it's bound to `127.0.0.53`, `dig localhost` will always return `127.0.0.1`.

**Review by Oats87 — DISMISSED** [review]

**Review by bfbachmann — APPROVED** [review]
LGTM! Very nice comments :)

**Review by Oats87 — APPROVED** [review]

**Pull request merged**